File: 455. Assign Cookies.cpp
class Solution {
public:
    // Greedy and sort: O(min(mlogM, nlogN))time, O(1)space, 63.37%, ok
    int findContentChildren(vector<int>& g, vector<int>& s) {
        if(g.empty() || s.empty()) return 0;
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int ret = 0, i = 0, j = 0;
        while(i < s.size() && j < g.size()){
            if(s[i] >= g[j]) {ret++; i++; j++;}
            else i++;
        }
        return ret;
    }
};

File: 455. Assign Cookies.py
class Solution(object):
    
    # sort and greedy: O(nlogn + mlogm + max(m,n)) time, O(1) space
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        ret = 0
        g.sort()
        s.sort()
        i, j = len(g) - 1, len(s) - 1
        while i >= 0 and j >= 0:
            if s[j] >= g[i]:
                ret += 1
                i, j = i - 1, j - 1
            else:
                i -= 1
        return ret
        
        